lambda/aws-qa-appsync-opensearch/question_answering/src/qa_agent/image_qa.py

```diff
@@ -0,0 +1,313 @@
+    image_url = get_presigned_url(bucket_name,filename)
+    if image_url is None:
+        logger.error(f'cannot find image url, returning..') 
+        status_variables['jobstatus'] = JobStatus.ERROR_LOAD_DOC.status
+        error = JobStatus.ERROR_LOAD_DOC.get_message()
+        status_variables['answer'] = error.decode("utf-8")
+        send_job_status(status_variables)
+        return status_variables
+
+    template = """\n\nUser: {question}![]({image})<end_of_utterance>
+         \n\nAssistant:"""
+    prompt = PromptTemplate(template=template, input_variables=["image", "question"])
+    chain = LLMChain(llm=_qa_llm, prompt=prompt, verbose=input_params['verbose'])
+    
+    try:    
+        logger.info(f'decoded_question is: {decoded_question}')
+        tmp = chain.predict(image=image_url, question=decoded_question)
+        answer=tmp.split("Assistant:",1)[1]
+        
+        logger.info(f'answer is: {answer}')
+        llm_answer_bytes = answer.encode("utf-8")
+        base64_bytes = base64.b64encode(llm_answer_bytes)
+        llm_answer_base64_string = base64_bytes.decode("utf-8")
+
+        status_variables['jobstatus'] = JobStatus.DONE.status
+        status_variables['answer'] = llm_answer_base64_string
+        
+    except Exception as err:
+        logger.exception(err)
+        status_variables['jobstatus'] = JobStatus.ERROR_PREDICTION.status
+        error = JobStatus.ERROR_PREDICTION.get_message()
+        status_variables['answer'] = error.decode("utf-8")
+    
+    return status_variables
+
+def generate_vision_answer_bedrock(bedrock_client,base64_images, model_id,decoded_question):
+    system_prompt=""
+    # use system prompt for fine tuning the performamce
+    # system_prompt= """
+    # You have perfect vision and pay great attention to detail which
+    # makes you an expert at answering architecture diagram question. 
+    # Answer question in <question></question> tags. Before answer, 
+    # think step by step in <thinking> tags and analyze every part of the diagram.
+    # """
+    #Create a prompt with the question
+    prompt =f"<question>{decoded_question}</question>. Answer must be a numbered list in a small paragraph inside <answer></answer> tag."
+
+    claude_config = {
+        'max_tokens': 1000, 
+        'temperature': 0, 
+        'anthropic_version': '',  
+        'top_p': 1, 
+        'stop_sequences': ['Human:']
+        }
+    
+    messages = {    
+        "role": "user",
+        "content": [
+            {
+                "type": "image",
+                "source": {
+                    "type": "base64",
+                    "media_type": "image/jpeg",
+                    "data": base64_images
+                }
+            },
+            {
+                "type": "text",
+                "text": prompt
+            
+            }
+        ]
+    }
+
+    body=json.dumps({'messages': [messages],**claude_config, "system": system_prompt})
+    response = bedrock_client.invoke_model(
```

Comment:
error handling around this call

Reply:
done - 7d4d3a152fa0f0b4a28117814347191758cae246